source/common/filter/config_discovery_impl.h

```diff
@@ -387,9 +392,41 @@ class FilterConfigProviderManagerImplBase : Logger::Loggable<Logger::Id::filter>
                                          absl::string_view type_url) const;
   void validateProtoConfigTypeUrl(const std::string& type_url,
                                   const absl::flat_hash_set<std::string>& require_type_urls) const;
+  virtual const std::string getConfigDumpType() const PURE;
 
 private:
+  void setupEcdsConfigDumpCallbacks(Server::Admin& admin) {
+    if (config_tracker_entry_ == nullptr) {
+      config_tracker_entry_ = admin.getConfigTracker().add(
+          getConfigDumpType(), [this](const Matchers::StringMatcher& name_matcher) {
+            return dumpEcdsFilterConfigs(name_matcher);
+          });
+    }
+  }
+
+  ProtobufTypes::MessagePtr dumpEcdsFilterConfigs(const Matchers::StringMatcher& name_matcher) {
+    auto config_dump = std::make_unique<envoy::admin::v3::EcdsConfigDump>();
+    for (const auto& subscription : subscriptions_) {
+      const auto& ecds_filter = subscription.second.lock();
```

Comment:
Maybe check that weak_ptr lock is non-null for defensive programming?

Reply:
Good idea. Done!